### PR TITLE
Test flakiness fixes - 2020-03-30 Edition

### PR DESCRIPTION
**Pull request by notnoop**
Best reviewed on commit by commit basis.

Running 55 workflows with the changes here, resulted into 30 failing builds with the following tests:

```
11      client/allocrunner/taskrunner/template/TestTaskTemplateManager_BlockedEvents
2       nomad/TestClientCSIController_AttachVolume_Forwarded
2       nomad/TestAutopilot_RollingUpdate
1       nomad/TestLeader_UpgradeRaftVersion
1       nomad/TestHeartbeat_Server_HeartbeatTTL_Failover
1       nomad/TestHeartbeat_ResetHeartbeatTimerLocked_Renew
1       nomad/TestDrainer_AllTypes_NoDeadline
1       nomad/TestClientFS_Logs_Remote_Server
1       nomad/TestClientAllocations_Stats_Remote
1       drivers/shared/executor/TestExecutor_Start_Wait/LibcontainerExecutor
1       drivers/shared/executor/TestExecutor_Start_Wait
1       api/TestEvaluations_Allocations
1       api/TestAllocations_ExecErrors
1       api/TestAgent_Datacenter
1       api/TestACLPolicies_Info
```

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -942,26 +942,6 @@ jobs:
 workflows:
   build-test:
     jobs:
-    - build-binaries:
```
Comment (tgross) [inline .circleci/config.yml]: Is this just a temporary removal for purposes of testing-out this PR more quickly?

File: nomad/drainer/watch_jobs.go [hunk 2]
```diff
@@ -144,13 +144,11 @@ func (w *drainingJobWatcher) watch() {
 	for {
 		w.logger.Trace("getting job allocs at index", "index", waitIndex)
 		jobAllocs, index, err := w.getJobAllocs(w.getQueryCtx(), waitIndex)
-		w.logger.Trace("retrieved allocs for draining jobs", "num_allocs", len(jobAllocs), "index", index, "error", err)
```
Comment (tgross) [inline nomad/drainer/watch_jobs.go]: 👍 I've run into this one a bunch in local development and hadn't had the chance to dig into it yet.

File: .circleci/config.yml (hunk 1 above)
Comment (notnoop) [reply .circleci/config.yml]: 🤦‍♂ Oops - that was only meant for the branch I checking for flakiness

File: nomad/client_agent_endpoint_test.go [hunk 3]
```diff
@@ -231,11 +232,13 @@ func TestMonitor_Monitor_RemoteServer(t *testing.T) {
 			require := require.New(t)
 
 			// send some specific logs
-			doneCh := make(chan struct{})
+			ctx, cancel := context.WithCancel(context.Background())
```
Comment (drewbailey) [inline nomad/client_agent_endpoint_test.go]: :+1: thanks

**Pull request merged**